app/controllers/provider_interface/offer/conditions_controller.rb

```diff
@@ -61,9 +72,15 @@ def submit_form(action:)
           @wizard.remove_empty_conditions!
           @wizard.save_state!
 
-          redirect_to [action, :provider_interface, @application_choice, :offer, @wizard.next_step]
+          respond_to do |format|
+            format.js { render js: "window.location.replace('#{polymorphic_url([action, :provider_interface, @application_choice, :offer, @wizard.next_step])}')" }
+            format.html { redirect_to [action, :provider_interface, @application_choice, :offer, @wizard.next_step] }
+          end
         else
-          render action
+          respond_to do |format|
+            format.js { render js: 'window.location.reload()' }
```

Comment:
This doesn't work:
- The HTML method relies on running the validations on the wizard, and then rendering the view with the already validated wizard object
- This just redirects to the new action, which makes a new (unvalidated) wizard.